some bug fixes

fcuny opened the pull request.

Commit:
Fix incorrect method name

ChangeDetailService don't have a method named `patchSetDetail`. Should be
calling `changeDetail` instead.

Commit:
Extend `changes` with advanced search parameters

It's possible to search for other things than a project name or status. The
`advanced_search` parameter allow to pass arbitrary parameters to search for:

    changes = client.changes(commit='f70f5e2dfa9476755')

or

    changes = client.changes(owner='fcuny')

For documentation about the advanced search, refere to
http://code.google.com/p/gerrit/issues/searchtips

Commit:
Fix method name to get patchset details

bcwaldon commented on gerrit/rpc.py:
```diff
@@ -34,7 +34,7 @@ def authenticate(self, method, **kwargs):
         raise error.UnknownAuthenticationMethodError()
         
-    def changes(self, status="open", project=None):
+    def changes(self, status="open", project=None, **advanced_search):
```
How about we treat project as part of advanced_search? And let's make advanced_search something shorter, just 'search' would work just as well.

fcuny replied:
yeah, that was also my original thinking, but I didn't want to introduce an incompatible change in the API without checking with you first. Of course, I forgot to ask this when I pushed my changes ...

Anyway, since we're OK on this, I'll update this change.

bcwaldon replied:
This library is in no way guaranteeing backwards compatibility at this level at the current time. Nobody is really using this just yet. I plan to refactor quite a bit from where it is now.

bcwaldon commented:
I don't mean to rush you, but I wanted to make sure there was another patch coming...

fcuny commented:
yes sir :) you can expect a patch later this evening or tomorrow

Commit:
Simplify the signature for `rpc.Client.changes`.

Removed `project` from the method signature and renamed `advanced_search` to
`search`.  `project` is can be a field in the `search` dict.  Tests are also
updated.

bcwaldon commented:
Sorry I let this sit so long! I need to get back on top of this project.
